src/vehicle_use.cpp

```diff
@@ -1582,6 +1582,97 @@ void vehicle::use_monster_capture( int part, map */*here*/, const tripoint_bub_m
     invalidate_mass();
 }
 
+void vehicle::use_tiedown_furniture( int part, map *here, const tripoint_bub_ms & )
+{
+    if( !here ) {
+        // should never happen!
+        debugmsg( "vehicle::use_tiedown_furniture() called without a map pointer, please report this error" );
+        return;
+    }
+
+    if( parts[part].is_broken() || parts[part].removed ) {
+        return;
+    }
+
+    const std::string item_var_string = "tied_down_furniture";
+
+    // Why do we construct a copy here?
+    item base_copy = item( parts[part].get_base() );
+
+    if( base_copy.has_var( item_var_string ) ) {
+
+        furn_str_id tied_down_furniture( base_copy.get_var( item_var_string ) );
+        if( !tied_down_furniture.is_valid() ) {
+            debugmsg( "Invalid stored furniture %s", base_copy.get_var( item_var_string ) );
+            base_copy.erase_var( item_var_string ); // to prevent bricking the item
+            return;
+        }
+
+        const std::function<bool( const tripoint_bub_ms & )> tile_without_furniture_obstruction =
+        [&here]( const tripoint_bub_ms & target ) {
+            // target does not have furniture or any vehicle part
+            // TODO: Where else can't we place a new furniture? There must be a common function for this
+            if( !here->has_furn( target ) && !here->veh_at( target ) ) {
+                return true;
+            }
+            return false;
+        };
+
+        // doesn't actually display for choose_adjacent_highlight since we don't allow auto selection
+        std::string fail_msg =
+            _( "Can't put furniture there.  There is a vehicle or another furniture in the way." );
+
+        std::optional<tripoint_bub_ms> selected_tile = choose_adjacent_highlight( *here,
+                _( "Select where to place the furniture." ),
+                fail_msg,
+                tile_without_furniture_obstruction, false, false );
+
+        if( selected_tile ) {
+            here->furn_set( *selected_tile, tied_down_furniture );
+            base_copy.erase_var( item_var_string );
+            parts[part].set_base( std::move( base_copy ) );
+            // FIXME: stored_furniture_weight is a bad hack, but furniture don't have json-defined weight
+            invalidate_mass();
+            return;
+        } else {
+            add_msg( fail_msg );
+            return; // user cancelled or picked invalid tile
+        }
+    }
+
+    // This is used to highlight only squares with moveable furniture
+    const std::function<bool( const tripoint_bub_ms & )> tile_with_furniture =
+    [&here]( const tripoint_bub_ms & target ) {
+        if( here->has_furn( target ) && here->furn( target )->move_str_req >= 0 ) {
```

Comment:
You can give clearer feedback if you let them select any adjacent tile, then back out if the tile they pick doesn't have furniture or has furniture that can't be moved.

Also consider checking the avatar's strength for this.

Reply:
I think choose_adjacent_highlight() is much better when dealing with few conditions, but if I do add a strength check that'll be 3 separate conditions - at that point it starts to warrant more specific messaging, as you say. So let me see about this.